.NET 6 Upgrade of Iot Connector Core Components

kenziedolish opened the pull request:
**Code Changes:**
Migrated the projects below to .NET 6
- Microsoft.Health.Fhir.Ingest.Console
- Microsoft.Health.Events
- Microsoft.Health.Extensions.Fhir
- Microsoft.Health.Extensions.Fhir.R4
- Microsoft.Health.Extensions.Host
- Microsoft.Health.Fhir.Ingest
- Microsoft.Health.Fhir.Ingest.Validation
- Microsoft.Health.Fhir.R4.Ingest
- Microsoft.Health.Common.UnitTests
- Microsoft.Health.Events.UnitTests
- Microsoft.Health.Expressions.UnitTests
- Microsoft.Health.Extensions.Fhir.UnitTests
- Microsoft.Health.Extensions.Fhir.R4.UnitTests
- Microsoft.Health.Fhir.Ingest.UnitTests
- Microsoft.Health.Fhir.Ingest.Template.UnitTests
- Microsoft.Health.Fhir.Ingest.Validation.UnitTests
- Microsoft.Health.Fhir.R4.Ingest.UnitTests

Upgraded Azure functions from v3 to v4 
- Microsoft.Health.Fhir.Ingest.Host

Updated Microsoft.Net.Compilers.Toolset from 3.9.0 -> 4.0.1
Updated Microsoft.NET.Sdk.Functions from 3.9.0 -> 4.0.1

Updated FUNCTIONS_EXTENSION_VERSION in arm templates to v4

_App Service running the Azure Function successfully_: [test-net6-upgrade-2](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/4092fd42-145e-4e91-aa15-96f66d2b6deb/resourceGroups/test-net6-upgrade-2/overview)

rogordon01 commented on src/lib/Microsoft.Health.Fhir.Ingest.Template/Microsoft.Health.Fhir.Ingest.Template.csproj:
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
We use this package inside of `health-pass`. AFAIK, `health-pass` still uses .net framework 4.7.2. The reason that matters is I don't think net472 supports .net core 6.0. We might need to keep this as netstandard2.0 so that it can continue to be used in `health-pass`.

https://dotnet.microsoft.com/en-us/platform/dotnet-standard

kenziedolish replied:
Connected offline with Rob about this - submitting an update soon

kenziedolish replied:
Sent out new commit that leaves all netstandard2.0 projects so that they can be used with net472 or .net6

rogordon01 commented:
The build is now broken. We'll need to update the pipeline:

`The current .NET SDK does not support targeting .NET 6.0.`

kenziedolish commented:
> The build is now broken. We'll need to update the pipeline.
> 
> `The current .NET SDK does not support targeting .NET 6.0.`
> 
> Most likely, we'll need to specify the new sdk inside of the build pipeline:
> 
> https://github.com/microsoft/iomt-fhir/blob/master/build/build.yml

Updated build.yml to use .NET Sdk 6.0.x and updated .NET.Sdk.Functions to v4

dustinburson commented on src/console/Microsoft.Health.Fhir.Ingest.Console.csproj:
```diff
@@ -1,36 +1,35 @@
 
-  <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Storage" Version="4.0.4" />
-    <PackageReference Include="Microsoft.Extensions.Hosting" Version="3.1.12" />
```
Looks like this reference was dropped.  Is it still needed?

kenziedolish replied:
No the Microsoft.Extensions.Hosting package was identified as transient during the migration so I removed it. I've also verified both the console app and the App Service running the Azure Functions are running successfully. 

Feel free to check it out: [test-net-upgrade-2](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/4092fd42-145e-4e91-aa15-96f66d2b6deb/resourceGroups/test-net6-upgrade-2/overview)

kenziedolish commented on src/console/Microsoft.Health.Fhir.Ingest.Console.csproj:
```diff
@@ -1,36 +1,35 @@
+			<PrivateAssets>all</PrivateAssets>
+			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+		</PackageReference>
+		<PackageReference Include="Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers" Version="0.3.261602">
```
No, I removed the Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzer package
